From: Hannes Reinecke <hare@suse.de>
Date: Mon, 14 Apr 2014 08:25:17 +0200
Subject: megaraid_sas: present JBOD devices to the OS
References: bnc#870440
Patch-Mainline: n/a

The fix for bug#698102 has the unfortunate side-effect to disallow
any physical devices to be presented to the OS.
This patch modifies the original patch so that JBODs are presented
again.

Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/megaraid/megaraid_sas_base.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/scsi/megaraid/megaraid_sas_base.c b/drivers/scsi/megaraid/megaraid_sas_base.c
index cea4cca..478d1f5 100644
--- a/drivers/scsi/megaraid/megaraid_sas_base.c
+++ b/drivers/scsi/megaraid/megaraid_sas_base.c
@@ -1656,9 +1656,6 @@ static int megasas_slave_configure(struct scsi_device *sdev)
 	 * system pd.
 	 */
 	if (sdev->channel < MEGASAS_MAX_PD_CHANNELS) {
-		if (sdev->type == TYPE_DISK)
-			return -ENXIO;
-
 		pd_index = (sdev->channel * MEGASAS_MAX_DEV_PER_CHANNEL) +
 								sdev->id;
 		if (instance->pd_list[pd_index].driveState !=
-- 
1.7.12.4

